src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerGeneralEmail.java

```diff
@@ -256,4 +219,177 @@ private boolean isEmailValid(Applicant applicant) {
+                        .value(DivorceDocument.builder().documentLink(uploadedDocumentOptional.get().getValue().getDocumentLink()).build())
+                        .build();
+                listOfAttachments.add(emailDoc);
+            }
+        }
+        addListToGeneralEmailAttachments(caseData, listOfAttachments);
+    }
+
+    void addSelectedGeneratedDocuments(final CaseData caseData) {
+        final GeneralEmail generalEmail = caseData.getGeneralEmail();
+
+        List<ListValue<DivorceDocument>> listOfAttachments = new ArrayList<>();
+
+        final List<DynamicListElement> selectedDocuments = generalEmail.getGeGeneratedDocumentNames().getValue();
+
+        for (DynamicListElement element : selectedDocuments) {
+            UUID uuidCode = element.getCode();
+            Optional<ListValue<DivorceDocument>> uploadedDocumentOptional =
+                emptyIfNull(caseData.getDocuments().getDocumentsGenerated())
+                    .stream()
+                    .filter(doc -> UUID.fromString(doc.getId()).equals(uuidCode))
+                    .findFirst();
+
+            if (uploadedDocumentOptional.isPresent()) {
+                ListValue<DivorceDocument> emailDoc =
+                    ListValue.<DivorceDocument>builder()
+                        .id(documentIdProvider.documentId())
+                        .value(DivorceDocument.builder().documentLink(uploadedDocumentOptional.get().getValue().getDocumentLink()).build())
+                        .build();
+                listOfAttachments.add(emailDoc);
+            }
+        }
+        addListToGeneralEmailAttachments(caseData, listOfAttachments);
+    }
+
+    void addSelectedApp1Documents(final CaseData caseData) {
+        GeneralEmail generalEmail = caseData.getGeneralEmail();
+
+        List<ListValue<DivorceDocument>> listOfAttachments = new ArrayList<>();
+
+        final List<DynamicListElement> selectedDocuments = generalEmail.getGeApplicant1DocumentNames().getValue();
+
+        for (DynamicListElement element : selectedDocuments) {
+            UUID uuidCode = element.getCode();
+            Optional<ListValue<DivorceDocument>> uploadedDocumentOptional =
+                emptyIfNull(caseData.getDocuments().getApplicant1DocumentsUploaded())
+                    .stream()
+                    .filter(doc -> UUID.fromString(doc.getId()).equals(uuidCode))
+                    .findFirst();
+
+            if (uploadedDocumentOptional.isPresent()) {
+                ListValue<DivorceDocument> emailDoc =
+                    ListValue.<DivorceDocument>builder()
+                        .id(documentIdProvider.documentId())
+                        .value(DivorceDocument.builder().documentLink(uploadedDocumentOptional.get().getValue().getDocumentLink()).build())
+                        .build();
+                listOfAttachments.add(emailDoc);
+            }
+        }
+        addListToGeneralEmailAttachments(caseData, listOfAttachments);
+    }
```

Comment:
To avoid duplicating these methods, I'm thinking we could refactor to have 1 method that accepts a collection of selected document names and a collection of case documents to compare against as arguments?

Reply:
Have added two functions addSelectedDivorceDocuments and addSelectedScannedDocuments to reuse rather than individual function to add eachh.